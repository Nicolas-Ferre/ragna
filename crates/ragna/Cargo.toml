[package]
name = "ragna"
description = "Library for easily creating GPU-native applications"
readme = "../../README.md"
keywords = ["gpu", "rendering", "compute", "shader"]
categories = ["game-development", "graphics", "rendering", "concurrency"]
exclude = [".github", "README.md"]
version.workspace = true
authors.workspace = true
edition.workspace = true
license.workspace = true
repository.workspace = true
rust-version.workspace = true

[dependencies]
bitfield-struct.workspace = true
derive-where.workspace = true
futures.workspace = true
fxhash.workspace = true
itertools.workspace = true
once_cell.workspace = true
wgpu.workspace = true
ragna_derive.workspace = true

[lints]
workspace = true
