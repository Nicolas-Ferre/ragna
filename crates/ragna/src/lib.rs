//! A library for creating easily a GPU-native application.

mod app;
mod operations;
mod operators;
mod runner;
mod types;
mod wgsl;

pub use app::*;
pub use operators::*;
pub use types::*;

/// Transforms a Rust module to a GPU module.
pub use ragna_derive::gpu;

// TODO: adapt compilation tests
