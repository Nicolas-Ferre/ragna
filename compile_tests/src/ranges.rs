fn main() {}

#[ragna::gpu]
mod gpu {
    use ragna::{Range, I32};

    static RANGE_WITHOUT_START: Range<I32> = ..5;
    static RANGE_WITHOUT_END: Range<I32> = 3..;
    static UNSUPPORTED_RANGE_FORM: Range<I32> = 3..=5;
}
