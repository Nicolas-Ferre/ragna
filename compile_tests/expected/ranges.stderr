error: missing bound start
 --> compile_tests/src/ranges.rs:7:46
  |
7 |     static RANGE_WITHOUT_START: Range<I32> = ..5;
  |                                              ^

error: missing bound end
 --> compile_tests/src/ranges.rs:8:44
  |
8 |     static RANGE_WITHOUT_END: Range<I32> = 3..;
  |                                            ^

error: unsupported range type
 --> compile_tests/src/ranges.rs:9:50
  |
9 |     static UNSUPPORTED_RANGE_FORM: Range<I32> = 3..=5;
  |                                                  ^

error[E0308]: mismatched types
 --> compile_tests/src/ranges.rs:7:46
  |
7 |     static RANGE_WITHOUT_START: Range<I32> = ..5;
  |                                              ^^^ expected `Range<I32>`, found `RangeTo<{integer}>`
  |
  = note: expected struct `ragna::Range<I32>`
             found struct `RangeTo<{integer}>`

error[E0308]: mismatched types
 --> compile_tests/src/ranges.rs:8:44
  |
8 |     static RANGE_WITHOUT_END: Range<I32> = 3..;
  |                                            ^^^ expected `Range<I32>`, found `RangeFrom<{integer}>`
  |
  = note: expected struct `ragna::Range<I32>`
             found struct `std::ops::RangeFrom<{integer}>`